cdc/kv/etcd.go

```diff
@@ -169,10 +179,34 @@ func (c CDCEtcdClient) SaveChangeFeedInfo(ctx context.Context, info *model.Chang
 	return errors.Trace(err)
 }
 
+// GetAllTaskPositions queries all task positions of a changefeed, and returns a map
+// mapping from captureID to TaskPositions
+func (c CDCEtcdClient) GetAllTaskPositions(ctx context.Context, changefeedID string, opts ...clientv3.OpOption) (map[string]*model.TaskPosition, error) {
+	key := GetEtcdKeyTaskPositionList(changefeedID)
+	resp, err := c.Client.Get(ctx, key, append([]clientv3.OpOption{clientv3.WithPrefix()}, opts...)...)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	positions := make(map[string]*model.TaskPosition, resp.Count)
```

Comment:
It seems `model.TaskPosition` is a tiny struct, we can use value instead of pointer for better performance?

Reply:
no, I think we should use pointer, If we try to update `ResolvedTS` or `CheckpointTS`, we should make sure that the new value will be set to `positionsMap`